src/D3MHub.sol

```diff
@@ -283,8 +284,10 @@ contract D3MHub {
         uint256 toUnwind;
         uint256 toWind;
 
-        // Determine if it needs to fully unwind due to D3M ilk being caged (but not culled) or plan is not active
-        if (ilks[ilk].tic != 0 || !ilks[ilk].plan.active()) {
+        // Determine if it needs to fully unwind due to D3M ilk being caged (but not culled), plan is not active or something
+        // wrong is going with the third party and we are entering in the ilegal situation of having less assets than registered
+        // It's adding up `WAD` due possible rounding errors
+        if (ilks[ilk].tic != 0 || !ilks[ilk].plan.active() || currentAssets + WAD < ink) {
```

Comment:
Instead of `WAD` I think we could just do 1 in the precision diff between DAI and the token. For Aave 1 wei should be enough, for Compound I think the margin should be 10^10. But doing that different per `ilk` would require ro add that number somehow in the `pool` or `plan` and I don't think it is probably worth it. 1 WAD is not much difference to really care.

Reply:
Other thing that realized and discussed with @talbaneth, adding up `WAD` means that a malicious value of `currentAssets > maxinuint256 - WAD` will make a potential unwinding to stop.
We agreed that caging or making the plan inactive (which won't require gov delay) it is probably more than enough protection for it (as the conditional will jump before than evaluating it). cc @kmbarry1

Reply:
Good edge case to call out but I doubt it would be hit in practice, and if someone is able to manipulate the `currentAssets` value (or if it has a bug) action would be necessary anyway. Don't think the side effects here are terribly severe, either.

Reply:
Yeah seems we are all align regarding this. Thanks for your feedback!